### ISSUE TITLE
[Bug] Error converting Keras LSTM model (TensorFlow 2)

### ISSUE THREAD

**Issue by n-Guard** [openvinotoolkit/openvino#5988]
##### System information (version)
- OpenVINO => 2021.3
- Operating System / Platform => macOS 11
- Problem classification: Model Conversion (tf.keras)
- Framework: TensorFlow => 2.3.2 (also tried 2.4.1)
- Python: 3.8

##### Detailed description
I was unable to convert a custom tf.keras LSTM model with only one layer to the OpenVINO format with the following error:

`[ ERROR ]  Cannot infer shapes or values for node "functional_1/lstm/PartitionedCall/TensorArrayV2_1".`
`[ ERROR ]  Tensorflow type 21 not convertible to numpy dtype.`
`[ ERROR ]  It can happen due to bug in custom shape infer function <function tf_native_tf_node_infer at 0x7fe733660670>.`
`[ ERROR ]  Or because the node inputs have incorrect values/shapes.`
`[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).`
`[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.`
`[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "functional_1/lstm/PartitionedCall/TensorArrayV2_1" node.`

##### Steps to reproduce
Execute the following script to convert a Keras LSTM model to a TensorFlow frozen graph:
https://gist.github.com/n-Guard/b446573a86b1c5d31382156ffc6e20ac

Run OpenVINO model optimizer script:
`python mo_tf.py --input_model frozen_tensorflow_lstm_model.pb --output_dir . --input_shape [1,10,15] --data_type FP32`

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by Iffa-Meah** [comment]
Hi, 

If possible can you share the specific model that you are having issue converting with?
Plus may I know what topology your model is? It's possible that the error occurred due to an unsupported topology.

**Comment by n-Guard** [comment]
Hi, I already defined the model in the gist I posted above, but here it is again:

`inp = Input(shape=(10, 15))`
`lstm = LSTM(5, return_sequences=True)(inp)`
`out = Dense(5, activation='softmax')(lstm)`
`model = Model(inputs=inp, outputs=out)`

**Comment by Iffa-Meah** [comment]
I checked in  Windows to see "in case" it works, but unfortunately it's not.
I've got the same error. This means that whether the topology is unsupported or something's wrong with your model.
Especially related with this [ ERROR ] It can happen due to bug in custom shape infer function <function tf_native_tf_node_infer at 0x7fe733660670>.

![K_convert](https://user-images.githubusercontent.com/79643476/121321461-b3689e80-c940-11eb-8cbe-f921b50d42d1.png)
![K_IR_Convert](https://user-images.githubusercontent.com/79643476/121321479-b82d5280-c940-11eb-9556-510926a8e48e.png)
